recipes/uvw/all/conanfile.py

```diff
@@ -13,73 +18,56 @@ class UvwConan(ConanFile):
-                self.settings.compiler.version)
-            )
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
 
     @property
-    def _required_EXACT_libuv_version(self):
-        """ Return *EXACT* libuv version to use with this uvw library """
+    def _required_libuv_version(self):
+        # uvw is bound to a particular libuv version which is part of the release archive name
         match = re.match(r".*libuv[_-]v([0-9]+\.[0-9]+).*", self.conan_data["sources"][self.version]["url"])
         if not match:
-            raise ConanException("uvw recipe does not know what version of libuv to use as dependency")
-        return tools.Version(match.group(1))
+            raise ConanException(f"{self.ref} does not know what version of libuv to use as a dependency")
+        return match.group(1)
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        libuv_version = self._required_EXACT_libuv_version
-        revision = 0
-        if libuv_version.major == "1" and libuv_version.minor == "44":
-            revision = 1
-        self.requires("libuv/{}.{}.{}".format(libuv_version.major, libuv_version.minor, revision))
+        self.requires(f"libuv/[~{self._required_libuv_version}]")
 
     def package_id(self):
-        self.info.header_only()
+        self.info.clear()
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("{self.ref} requires C++{self._min_cppstd}, which your compiler doesn't support")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def package(self):
-        self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "src"))
-        self.copy("*", dst=os.path.join("include", "uvw"), src=os.path.join(self._source_subfolder, "src", "uvw"))
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        copy(self, "*.hpp", src=os.path.join(self.source_folder, "src"), dst=os.path.join(self.package_folder, "include"))
+        copy(self, "*", src=os.path.join(self.source_folder, "src", "uvw"), dst=os.path.join(self.package_folder, "include", "uvw"))
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
 
     def package_info(self):
-        # Check whether the version of libuv used as a requirement is ok
```

Comment:
I removed the additional version check in package_info because a correct version of libuv will be enforced as part of the normal requirements resolution in Conan.

I think it should be allowed to override the version for flexibility. From my experience, uvw often works with a slightly different version of libuv. If it doesn't then the conan recipe is not to blame.